debezium-core/src/main/java/io/debezium/transforms/tracing/ActivateOpenTracingSpan.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.transforms.tracing;
+
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.data.Envelope;
+import io.debezium.transforms.SmtManager;
+import io.opentracing.Scope;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.Tracer.SpanBuilder;
+import io.opentracing.propagation.Format;
+import io.opentracing.util.GlobalTracer;
+
+/**
+ * This SMT enables integration with an OpenTracing system.
+ * The SMT creates a OpenTracing span and enriches it with metadata from envelope and source info block.<br/>
+ * It is possible to connect the span to a parent span created by a business application.
+ * The application then needs to export its OpenTracing active span into a database field.
+ * The SMT looks for a predefined field name in the {@code after} block
+ * and when found it extracts the parent span from it.<br>
+ * 
+ * See {@link EventDispatcher} for example of such implementation.
+ * 
+ *
+ * @param <R> the subtype of {@link ConnectRecord} on which this transformation will operate
+ * @author Jiri Pechanec
+ */
+public class ActivateOpenTracingSpan<R extends ConnectRecord<R>> implements Transformation<R> {
```

Comment:
I don't quite understand the advantage of that separate SMT yet, when not using the outbox pattern. Won't the producer interceptor effecitively do the same?

That said, assuming for a second the separate SMT is useful, I'd still prefer this to be more unified experience for the user of the outbox routing SMT. I.e. they should have only to use a single SMT (the router) and be done. It may imply some duplication/complexity for ourselves, but it's easier for the user.

Reply:
@gunnarmorling No it won't. The intereceptor will get the message after it is serialized so it is opaque. With SMT we work on logical message so we can extract the Debezium metadata (source info).

Reply:
Yes, got it now. That make sense.

My point stands though that I'd prefer an outbox user doesn't have to deal with two separate SMTs.

Reply:
@gunnarmorling IMHO I'd just make EventRouter to call `ActivateOpenTracingSpan` SMT automatically so it will not be necessary to add it to the transforms list.

Reply:
That sounds great!